frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -257,38 +260,31 @@ impl<T: Config> Pallet<T> {
 	/// If you want a checked solution and submit it at the same time, use
 	/// [`Pallet::mine_check_save_submit`].
 	pub fn mine_and_check(
-		iters: usize,
-	) -> Result<(RawSolution<SolutionOf<T>>, SolutionOrSnapshotSize), MinerError> {
-		let (raw_solution, witness) = Self::mine_solution(iters)?;
+	) -> Result<(RawSolution<SolutionOf<T>>, SolutionOrSnapshotSize), MinerError<T>> {
+		let (raw_solution, witness) = Self::mine_solution::<T::Solver>()?;
 		Self::basic_checks(&raw_solution, "mined")?;
 		Ok((raw_solution, witness))
 	}
 
 	/// Mine a new npos solution.
-	pub fn mine_solution(
-		iters: usize,
-	) -> Result<(RawSolution<SolutionOf<T>>, SolutionOrSnapshotSize), MinerError> {
+	pub fn mine_solution<S: NposSolver<AccountId = T::AccountId, Error = SolverErrorOf<T>>>(
+	) -> Result<(RawSolution<SolutionOf<T>>, SolutionOrSnapshotSize), MinerError<T>> {
 		let RoundSnapshot { voters, targets } =
 			Self::snapshot().ok_or(MinerError::SnapshotUnAvailable)?;
 		let desired_targets = Self::desired_targets().ok_or(MinerError::SnapshotUnAvailable)?;
 
-		seq_phragmen::<_, SolutionAccuracyOf<T>>(
-			desired_targets as usize,
-			targets,
-			voters,
-			Some((iters, 0)),
-		)
-		.map_err(Into::into)
-		.and_then(Self::prepare_election_result)
+		S::solve(desired_targets as usize, targets, voters)
+			.map_err(|e| MinerError::Solver::<T>(e))
+			.and_then(|e| Self::prepare_election_result::<S>(e))
 	}
 
 	/// Convert a raw solution from [`sp_npos_elections::ElectionResult`] to [`RawSolution`], which
 	/// is ready to be submitted to the chain.
 	///
 	/// Will always reduce the solution as well.
-	pub fn prepare_election_result(
-		election_result: ElectionResult<T::AccountId, SolutionAccuracyOf<T>>,
-	) -> Result<(RawSolution<SolutionOf<T>>, SolutionOrSnapshotSize), MinerError> {
+	pub fn prepare_election_result<S: NposSolver>(
+		election_result: ElectionResult<T::AccountId, S::Accuracy>,
```

Comment:
```suggestion
	pub fn prepare_election_result(
		election_result: ElectionResult<T::AccountId, SolutionAccuracyOf<T>>,
```

``This function can only convert some `ElectionResult` that has our desired small-ish accuracy (`SolutionAccuracyOf`, which is `PerU16`)